File: day6/solution.js
const fs = require("fs");

let data = fs.readFileSync("input.txt", "utf8");
const [signal] = data.split("\n").filter((item) => item);

const findSignal = (n) => {
  for (let i = n; i < signal.length; i++) {
    const subString = signal.slice(i - n, i);
    if ([...new Set(subString)].length === n) return i;
  }
};

console.log("Question 1: ", findSignal(4));
console.log("Question 2: ", findSignal(14));

File: day6/solution.py
with open('input.txt', 'r') as f:
    lines = f.readlines()

signal = lines[0].strip()

# Question 1:
for i in range(len(signal)):
    if len(set(signal[i: i+4])) == 4:
        print("Question 1: ", i+4)
        break

# Question 2:
for i in range(len(signal)):
    if len(set(signal[i: i+14])) == 14:
        print("Question 2: ", i+14)
        break

